addons/mail/models/mail_message.py

```diff
@@ -11,6 +11,7 @@
 
 
 _logger = logging.getLogger(__name__)
+_image_dataurl = re.compile(r'(data:image/(.+));base64,([^"\']*)', re.I)
```

Comment:
`.+` is going to backtrack a ton (especially as the data part tends to be really long), completely unnecessarily. It should be a non-greedy qualifier (`.+?`) since the subtype should be very short compared to the complete document size (on a small test, the greedy version consistently takes ~twice as long as the non-greedy ones).

And considering the context, `[a-z]+?` or `\w+?` would probably be a better idea as we're specifically parsing images *and* we probably don't care about image/x-icon or image/svg+xml. If we want to apply RFC6838 strictly, it should be `[a-z0-9][-!#$&^_.+a-z0-9]{,126}`, though I don't think that really matters.